### PR TITLE
add log file name format configure

### PR DESCRIPTION
**Pull request by wenxuwan**
add log name format configure in global_log_roller.

eg : User can configure the log name with "format=mosn-2006-01-02_15.log" 

image

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -140,9 +143,10 @@ func GetOrCreateLogger(output string, roller *Roller) (*Logger, error) {
 	if roller.Handler == nil {
 		roller.Handler = rollerHandler
 	}
-
+	outputPath := output[0 : strings.LastIndex(output, "/")+1]
```
Comment (nejisama) [inline log/logger.go:146]: if output is "", a panic occured

**Comment by nejisama** [comment]
The config makes the log file name changed. that makes confused.

File: log/logger_test.go [hunk 2]
```diff
@@ -330,3 +330,94 @@ func TestRotateRightNow(t *testing.T) {
...
 	}
 }
+
+func TestRotateWithFormat(t *testing.T) {
+	err := InitGlobalRoller("format=mosn-2006-01-02_15.log")
+	if err != nil {
+		t.Fatal(err)
+	}
+	logName := "/tmp/mosn_bench/TestRotateWithFormat.log"
+	rollerName := "/tmp/mosn_bench/" + "mosn-" + time.Now().Format("2006-01-02_15") + ".log"
```
Comment (nejisama) [inline log/logger_test.go:340]: after rotate, `TestRotateWithFormat.log` changed to another name `mosn-xxx`.
if there are two loggers or more, the log will be lose

**Comment by wenxuwan** [comment]
> The config makes the log file name changed. that makes confused.

Yes, i will close the pr, thanks for your comments

**Pull request closed without merging**